pymc/distributions/distribution.py

```diff
@@ -622,13 +682,38 @@ def dist(
         if logcdf is None:
             logcdf = default_not_implemented(class_name, "logcdf")
 
-        if moment is None:
-            moment = functools.partial(
-                default_moment,
-                rv_name=class_name,
-                has_fallback=True,
+        def dist_moment(rv, size, *dist_params):
```

Comment:
Why can't you use the fgraph of the `rv` here? I don't follow

Reply:
Here is what I have in mind:

```python
def dist_moment(rv, size, *dist_params):
  fgraph = rv.owner.op.fgraph.clone()
  replace_moments = MomentRewrite()
  replace_moments.rewrite(fgraph)
  # Note: You can use `fgraph.replace_all` here
  for i, par in enumerate([size] + list(dist_params)):
      fgraph.replace(fgraph.inputs[i], par)
  [moment] = fgraph.outputs
  return moment
```

Reply:
@ricardoV94 because I got
```
>>> rv
*** Not yet returned!
```

Reply:
I don't know what Not yet returned means?

Reply:
@ricardoV94 I just used pdb wrong, we do not need re-create rv here.